### PR TITLE
Change default PushType; remove MongoConfig and rely on Push inputs

### LINKED ISSUE #93: Mongo Adapter default Replace behaviour
#### Description:
Per conversation with @alelom and @epignatelli regarding Mongo_Adapter actionConfig default behaviors. By default the Mongo_Adapter will overwrite any existing data within a collection. I would argue this behavior is not intuitive and may lead to confusion since the MongoConfig input param 'Replace' is set to False by default. 

#### Steps to reproduce:
Push any data to be appended to any collection to MongoDB without using Mongo actionConfig.

#### Expected behaviour:
Expect to append data within collection, or at least concurrence between default adapter functionality and MongoConfig.

### LINKED ISSUE #150: Remove MongoConfig as its parameters are redundant with the `Push()` inputs
#### Broken rules:

While looking at #93, I realized that the MongoConfig currently specifies a `tag` property that is used in the Push to replace the tag input, which makes no sense.

It also specifies a `replace` boolean that does the functionality of the `PushType` input (available to the Push() method), which we should leverage instead.

#### Suggestions to restore compliance:

Delete the MongoConfig and refactor the Push code to leverage existing inputs.

### PR DESCRIPTION
**Pull request by alelom**
### Issues addressed by this PR

Closes #93
Closes #150 

The default Push behaviour is now to "append", while you need to set the `PushType` to `DeleteThenCreate` to "replace". The MongoConfig has been deleted, as it's redundant.

### Test files

Try any existing script, e.g. https://burohappold.sharepoint.com/:f:/s/BHoM/ElPtUXgB2_pEvRhg7WfRuvEBlaP93pGBZ_zPeWvWpTYtYQ?e=X8HQEI

### Changelog

- The default `Push` behaviour is now to "append"; you need to set the `PushType` to `DeleteThenCreate` to "replace", and any other `PushType` results in appending. 
- The [MongoConfig ](https://github.com/BHoM/Mongo_Toolkit/blob/9ae4c4690d7c679470b90f2583f45687d10e65a5/Mongo_oM/Adapter/MongoConfig.cs#L29-L34) has been deleted as redundant. When `Push`ing, the `tag` is accessible through the Push inputs, and the behaviour obtained through the `replace` boolean can be replicated by setting the `PushType`. When doing `Execute`, the `replace` boolean in the MongoConfig was already not used, as it was substituted for [the boolean available in the `Transfer` command](https://github.com/BHoM/Mongo_Toolkit/blob/9ae4c4690d7c679470b90f2583f45687d10e65a5/Mongo_oM/Commands/Transfer.cs#L35) a while ago.

### CODE REVIEW

**Comment by alelom** [comment]
@BHoMBot check required

**Comment by alelom** [comment]
@BHoMBot check required

File: Mongo_Adapter/AdapterActions/Push.cs [hunk 1]
```diff
@@ -67,6 +59,12 @@ public override List<object> Push(IEnumerable<object> objects, string tag = "",
             else
                 m_Collection.InsertMany(documents);
 
+            if (pushType != PushType.AdapterDefault && pushType != PushType.DeleteThenCreate && pushType != PushType.CreateOnly)
+                BH.Engine.Reflection.Compute.RecordNote($"{this.GetType().Name} only supports the following {nameof(PushType)}s:" +
+                    $"\n\t- {nameof(PushType.CreateOnly)} => appends content (default setting)" +
+                    $"\n\t- {nameof(PushType.DeleteThenCreate)} => replaces all content" +
+                    $"\nevery other {nameof(PushType)} will behave as {nameof(PushType.CreateOnly)}.");
```
Comment (jamesramsden-bh) [inline Mongo_Adapter/AdapterActions/Push.cs:66]:
```suggestion
                    $"\nEvery other {nameof(PushType)} will behave as {nameof(PushType.CreateOnly)}.");
```

**Review by jamesramsden-bh — DISMISSED** [review]
Works perfectly. Just tested on my current project, all fine.

**Review by jamesramsden-bh — APPROVED** [review]

**Comment by alelom** [comment]
@BHoMBot check required
@BHoMBot check ready-to-merge

**Comment by alelom** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check dataset-compliance
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
Having discussed with @alelom and am merging to continue testing via alphas over the weekend/early next week. Things we want to ensure are still working with this update will include analytics scraping, bot operations, and other existing uses of Mongo Toolkit in operation. FYI @adecler @michaelhoehn @IsakNaslundBh

**Pull request merged**